open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/k-means-clustering-with-python.ipynb

```diff
@@ -0,0 +1 @@
achieved a very weak classification accuracy of 1%."]},{"cell_type":"markdown","metadata":{},"source":["I will check the model accuracy with different number of clusters."]},{"cell_type":"markdown","metadata":{},"source":["## K-Means model with different clusters"]},{"cell_type":"markdown","metadata":{},"source":["### K-Means model with 3 clusters"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["kmeans = KMeans(n_clusters=3, random_state=0)\n","\n","kmeans.fit(X)\n","\n","# check how many of the samples were correctly labeled\n","labels = kmeans.labels_\n","\n","correct_labels = sum(y == labels)\n","print(\"Result: %d out of %d samples were correctly labeled.\" % (correct_labels, y.size))\n","print('Accuracy score: {0:0.2f}'. format(correct_labels/float(y.size)))"]},{"cell_type":"markdown","metadata":{},"source":["### K-Means model with 4 clusters"]},{"cell_type":"code","execution_count":null,"metadata":{"trusted":true},"outputs":[],"source":["kmeans = KMeans(n_clusters=4, random_state=0)\n","\n","kmeans.fit(X)\n","\n","# check how many of the samples were correctly labeled\n","labels = kmeans.labels_\n","\n","correct_labels = sum(y == labels)\n","print(\"Result: %d out of %d samples were correctly labeled.\" % (correct_labels, y.size))\n","print('Accuracy score: {0:0.2f}'. format(correct_labels/float(y.size)))"]},{"cell_type":"markdown","metadata":{},"source":["We have achieved a relatively high accuracy of 62% with k=4."]},{"cell_type":"markdown","metadata":{},"source":["## 17. Results and conclusion\n","\n","\n","1.\tIn this project, I have implemented the most popular unsupervised clustering technique called K-Means Clustering.\n","\n","2.\tI have applied the elbow method and find that k=2 (k is number of clusters) can be considered a good number of cluster to cluster this data.\n","\n","3.\tI have find that the model has very high inertia of 237.7572. So, this is not a good model fit to the data.\n","\n","4.\tI have achieved a weak classification accuracy of 1% with k=2 by our unsupervised model.\n","\n","5.\tSo, I have changed the value of k and find relatively higher classification accuracy of 62% with k=4.\n","\n","6.\tHence, we can conclude that k=4 being the optimal number of clusters.\n"]},{"cell_type":"markdown","metadata":{},"source":["## Acknowledgments\n","Thanks to Gokulprasanth T for creating the open-source project [K-Means Clustering with Python](https://www.kaggle.com/code/prashant111/k-means-clustering-with-python), lisensed under the [Prashant Banerjee](https://www.kaggle.com/prashant111). It inspires the majority of the content in this chapter."]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.17"}},"nbformat":4,"nbformat_minor":4}
```

Comment:
<p>The serial number "1." is not needed, and all the following serial numbers are the same, so delete them.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ocademy-ai/machine-learning/pull/598/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>